Automattic/jetpack
# Assets: Add JSX runtime polyfill to prevent issues with React's automatic runtime being used to transform jsx

Fixes https://github.com/Automattic/jetpack/issues/38424

## Proposed changes:

* This PR adds a polyfill for the react-jsx-runtime script, which will allow us to update the `dependency-extraction-webpack-plugin` package or add any other relevant package that uses React's automatic runtime to transform JSX, as it prevents breakages caused by old JSX transforms used throughout the codebase.
* It also removes a recently added script to prevent updating the above mentioned package.

### Other information:


## Jetpack product discussion

https://github.com/Automattic/jetpack/issues/38424

## Does this pull request change what data or activity we track or use?

No.


## Testing instructions:

To test, you'll need WordPress 6.5.5 active on a test site.

* If you're testing locally and using Docker, download WordPress 5.5 and then replace `tools/docker/wordpress/wp-admin`, and `tools/docker/wordpress/wp-includes`. Then replace `tools/docker/wordpress/wp-settings.php`. I found this was enough to be able to replicate the issue and test the fix.
* You'll also need to make sure your site doesn't include a plugin which also uses the polyfill, and Gutenberg will need to be deactivated. So best to keep it simple with plugins.

To replicate the issue:

* Apply this patch, and comment out the changes in `webpack.config.js` and `class-assets.php`.
* Also, search for all instances of `"@wordpress/dependency-extraction-webpack-plugin"` (there should be 6), and change the version to 6.0.0.
* Then run `jetpack build --deps plugins/jetpack`.
* Visit `/wp-admin/admin.php?page=my-jetpack`. The main My Jetpack UI doesn't load anything.
* Visit `/wp-admin/admin.php?page=jetpack#/dashboard`. The main content area doesn't load anything.
* Other plugins to test with that display broken / no UI:
  * Jetpack Social plugin: `/wp-admin/admin.php?page=jetpack-social`
  * Automattic for Agencies: `/wp-admin/options-general.php?page=automattic-for-agencies-client`
  * Jetpack Backup: `/wp-admin/admin.php?page=jetpack-backup`
  * Jetpack Search: `/wp-admin/admin.php?page=jetpack-search`
  * Jetpack VideoPress: `/wp-admin/admin.php?page=jetpack-videopress`

To test the fix:

* Un-comment the previously commented out changes, and re-build. 
* Both of those Jetpack pages should load properly, as should the admin pages for the other plugins mentioned above.
